core/string/ustring.cpp

```diff
@@ -628,12 +633,7 @@ String &String::operator+=(char32_t p_char) {
 
 bool String::operator==(const char *p_str) const {
 	// compare Latin-1 encoded c-string
-	int len = 0;
-	const char *aux = p_str;
-
-	while (*(aux++) != 0) {
-		len++;
-	}
+	int len = strlen(p_str);
```

Comment:
I think this is a bad replacement to do, because here your parameter is a `char *` but the strlen function expects a `char32_t *`, which means it will check for a character every 4 bytes, instead of every byte.

Reply:
This is calling the std::strlen overload (through `using namespace std`) which does the same thing, but for `char*`. Our `char32_t*` overload cannot be called with a `char*` parameter because the pointer cannot be implicitly converted. :)